tests/cstor/snapshot/snapshot_test.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package snapshot
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	snap "github.com/openebs/maya/pkg/kubernetes/snapshot/v1alpha1"
+	sc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	spc "github.com/openebs/maya/pkg/storagepoolclaim/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[cstor] TEST SNAPSHOT PROVISIONING", func() {
+	var (
+		err      error
+		pvcName  = "cstor-pvc"
+		snapName = "cstorsnapshot"
+	)
+
+	BeforeEach(func() {
+		When("we are creating pool deployment", func() {
+			By("building object of storageclass")
+			scObj, err = sc.NewBuilder().
```

Comment:
What namespace is being used to create this SC?
Can you please refer to comments provided at https://github.com/openebs/maya/pull/1215

Reply:
SC is clusterscope resource. :)

Reply:
i am using namespace `test-cstor-snap` and suffix the required resource names like storageclass is `test-cstor-snap-sc`  and pvc will be `test-cstor-snap-pvc` etc.

Reply:
ok.. in next PR have the cluster scope names auto generated. The suffix should be some auto generated thing.
make use of `metadata.generateName`